app/src/main/java/org/simple/clinic/drugs/selectionv2/entry/CustomPrescriptionEntrySheetv2.kt

```diff
@@ -0,0 +1,105 @@
+package org.simple.clinic.drugs.selectionv2.entry
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.inputmethod.EditorInfo
+import android.widget.Button
+import com.google.android.material.textfield.TextInputEditText
+import com.jakewharton.rxbinding2.view.RxView
+import com.jakewharton.rxbinding2.widget.RxTextView
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers.mainThread
+import io.reactivex.schedulers.Schedulers.io
+import io.reactivex.subjects.PublishSubject
+import kotterknife.bindView
+import org.simple.clinic.R
+import org.simple.clinic.activity.TheActivity
+import org.simple.clinic.widgets.BottomSheetActivity
+import org.simple.clinic.widgets.UiEvent
+import org.simple.clinic.widgets.textChanges
+import java.util.UUID
+import javax.inject.Inject
+
+class CustomPrescriptionEntrySheetv2 : BottomSheetActivity() {
+
+  private val drugNameEditText by bindView<TextInputEditText>(R.id.customprescription_drug_name)
+  private val drugDosageEditText by bindView<TextInputEditText>(R.id.customprescription_drug_dosage)
+  private val saveButton by bindView<Button>(R.id.customprescription_save)
+
+  @Inject
+  lateinit var controller: CustomPrescriptionEntryControllerv2
+
+  private val onDestroys = PublishSubject.create<Any>()
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.sheet_custom_prescription_entry_v2)
+    TheActivity.component.inject(this)
+
+    Observable.mergeArray(sheetCreates(), drugNameChanges(), drugDosageChanges(), drugDosageFocusChanges(), saveClicks())
+        .observeOn(io())
+        .compose(controller)
+        .observeOn(mainThread())
+        .takeUntil(onDestroys)
+        .subscribe { uiChange -> uiChange(this) }
+  }
+
+  override fun onDestroy() {
+    onDestroys.onNext(Any())
+    super.onDestroy()
+  }
+
+  override fun onBackgroundClick() {
+    val drugNameEmpty = drugNameEditText.text.isNullOrEmpty()
+    val dosageEmpty = drugDosageEditText.text.isNullOrEmpty()
+        || drugDosageEditText.text.toString().trim() == getString(R.string.customprescription_dosage_placeholder)
+
+    if (drugNameEmpty && dosageEmpty) {
+      super.onBackgroundClick()
+    }
+  }
+
+  private fun sheetCreates(): Observable<UiEvent> {
+    val patientUuid = intent.getSerializableExtra(KEY_PATIENT_UUID) as UUID
+    return Observable.just(CustomPrescriptionSheetCreated(patientUuid))
+  }
+
+  private fun drugNameChanges() = drugNameEditText.textChanges(::CustomPrescriptionDrugNameTextChanged)
```

Comment:
@sanchita I'm trying to review this PR commit-wise. I'm on the first commit and it looks like this commit is doing multiple things:
1. Adding V2 classes so that the new feature can be put behind a flag
2. Adding other changes like this one:

```kotlin
70: private fun drugDosageChanges() = drugDosageEditText.textChanges(::CustomPrescriptionDrugDosageTextChanged)
```

Because the entire file was duplicated, it's now difficult to track what was changed.

Reply:
This change is only because of the layout changes.
I have changed `EditText` -> `TextInputEditText`

Reply:
That shouldn't affect the existing code. They're both `EditText`, right?

Reply:
Additionally, you're not even making use of any APIs of `TextInputLayout`. You could continue using `EditText` as the type for `drugNameEditText` and `drugDosageEditText` while keeping `TextInputEditText` in the xml.